server.go

```diff
@@ -602,13 +752,18 @@ func (s *Server) Stop() {
 	s.lis = nil
 	cs := s.conns
 	s.conns = nil
+	for c := range s.hconns {
+		go c.Close()
```

Comment:
Some net.Conn impls block unreasonably long on Close. Until Go 1.6, even Go's *tls.Conn could hang forever in Close if there was a blocking Write in-flight (which wouldn't be woken up by the Close, since the Close was previously blocked forever on a mutex)

Spinning up a goroutine is nearly free.